### PR TITLE
intel-oneapi 2024.1.0: added new version to packages

### PR DESCRIPTION
**Pull request by prstrnn**

### CODE REVIEW

**Review by rscohn2 — DISMISSED** [review]
Thanks!

**Review by tldahlgren — DISMISSED** [review]
FYI. Confirmed all of the version sha256s.

**Comment by rscohn2** [comment]
Test is failing because MLK 2024.1 needs the 2024.1 compiler. e4s can update the compiler in its docker images after intel releases the 2024.1 docker images.
@eugeneswalker

File: var/spack/repos/builtin/packages/intel-oneapi-advisor/package.py [hunk 1]
```diff
@@ -24,6 +24,12 @@ class IntelOneapiAdvisor(IntelOneApiLibraryPackageWithSdk):
         "https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/advisor.html"
     )
 
+    version(
+        "2024.1.0",
+        url="https://registrationcenter-download.intel.com/akdlm/IRC_NAS/414cea14-4f3c-45f0-b854-44fb6cf9f34b/l_oneapi_advisor_p_2024.1.0.500_offline.sh",
+        sha256="1c327777a34a7e70e5840b9555ebf44615bf0295fcf3c673576d36a9a8979090",
+        expand=False,
+    )
```
Comment (rscohn2) [inline var/spack/repos/builtin/packages/intel-oneapi-advisor/package.py:32]: If you add `preferred=True` to the 2024.0.1 below then e4s would get past the problem with hefft in the e4s test. We can remove it after e4s has released docker images with the 2024.1 compilers

Comment (prstrnn) [reply var/spack/repos/builtin/packages/intel-oneapi-advisor/package.py:32]: @rscohn2  in Advisor? MKL test failed, so I confused a little bit. But if you confirm that I need to change advisor, I'll do that. It's fine to me

**Review by rscohn2 — APPROVED** [review]
Right, MKL.

**Pull request merged**